Sindri-Labs/sindri-scroll-sdk
# reorganize code for testing purposes

## What type of PR is this? (check all applicable)


## Related Tickets & Documents



- Closes #22

## Description

Move code out of main.rs into other files for ease of testing. According to Rust [test organization - integration tests for binary crates](https://doc.rust-lang.org/book/ch11-03-test-organization.html#integration-tests-for-binary-crates) a binary crate with all code in main.rs is not testable.

Break out most code into prover.rs.
Keep a simple `fn main` in main.rs.
Place public references in lib.rs.

## Developer Notes

## Testing Instructions, Screenshots, Recordings

`cargo run --release` and verify proper operation

_Please replace this line with instructions on how to test your changes, as well as any relevant
images for UI changes._

## Added/updated tests?
      have not been included_

## [optional] What gif best describes this PR or how it makes you feel?

![I think we should do a test](https://giphy.com/gifs/season-7-the-simpsons-7x4-3o6Mbbs879ozZ9Yic0)
